lexicon/providers/misaka.py

```diff
@@ -0,0 +1,229 @@
+
+
+def provider_parser(subparser):
+    """Configure provider parser for Misaka.IO"""
+    subparser.add_argument(
+        "--auth-token", help="specify token for authentication")
+
+
+class Provider(BaseProvider):
+    """Provider class for Misaka.IO"""
+    def __init__(self, config):
+        super(Provider, self).__init__(config)
+        self.domain_id = None
+        self.api_endpoint = 'https://api.misaka.io/dns'
+
+    def _authenticate(self):
+        payload = self._get(f'/zones/{self.domain}/settings')
+        if not payload['id']:
+            raise Exception('No domain found')
+
+        self.domain_id = self.domain
+
+    # Create or update a record.
+    def _create_record(self, rtype, name, content):
+        name = self._relative_name(name)
+        print(name)
+
+        identifier = f'{self.domain_id}/recordsets/{name}/{rtype}'
+        endpoint = f'/zones/{identifier}'
+        ttl = self._get_lexicon_option('ttl')
+        record = {'value': content}
+
+        existing_recordset = self._get_recordset(name, rtype)
+        if existing_recordset:
+            # append if returned recordsets doesn't include the record
+            if not _recordset_has_record(existing_recordset, content):
+                existing_recordset['records'].append(record)
+                existing_recordset['ttl'] = ttl
+                self._put(endpoint, existing_recordset)
+                LOGGER.debug('recordset exists, appending: %s', identifier)
+            # update ttl if returned ttl doesn't match
+            elif existing_recordset['ttl'] != ttl:
+                existing_recordset['ttl'] = ttl
+                self._put(endpoint, existing_recordset)
+                LOGGER.debug('recordset exists, updating ttl: %s', identifier)
+        else:
+            recordset = {
+                'ttl': ttl,
+                'filters': [],
+                'records': [record],
+            }
+            self._post(endpoint, recordset)
+            LOGGER.debug('create_record: %s', identifier)
+
+        return True
+
+    def _list_records(self, rtype=None, name=None, content=None):
+        params = {}
+        if name:
+            name = self._relative_name(name)
+            params['name'] = name
+
+        payload = self._get(f'/zones/{self.domain_id}/recordsets', query_params=params)
+        records = []
+        for recordset in payload['results']:
+            if not recordset['records']:
+                continue
+
+            if name and recordset['name'] != name:
+                continue
+
+            if rtype and recordset['type'] != rtype:
+                continue
+
+            processed_recordset = {
+                'name': self._full_name(recordset['name']),
+                'type': recordset['type'],
+                'ttl': recordset['ttl'],
+
+                # id is useless here as we don't have record ID exposed to customers
+                'id': '{}/recordsets/{name}/{type}'.format(self.domain_id, **recordset),
```

Comment:
Nit: you are using f-strings above, since Lexicon supports only Python 3.6+. Maybe you want to take advantage of it here also ?